test_scripts/API/VehicleData/commonVehicleData.lua

```diff
@@ -391,4 +557,75 @@ function m.processGetVDwithCustomDataSuccess()
   m.getMobileSession():ExpectResponse(cid, mobResParams)
 end
 
+function m.hmiLeveltoLimited(pAppId, pSystemContext)
+  pAppId = pAppId or 1
+  pSystemContext = pSystemContext or "MAIN"
+  m.getHMIConnection(pAppId):SendNotification("BasicCommunication.OnAppDeactivated",
+    { appID = m.getHMIAppId(pAppId) })
+  m.getMobileSession(pAppId):ExpectNotification("OnHMIStatus",
+    { hmiLevel = "LIMITED", audioStreamingState = "AUDIBLE", systemContext = pSystemContext })
+end
+
+function m.hmiLeveltoBackground(pSystemContext)
+  pSystemContext = pSystemContext or "MAIN"
+  m.getHMIConnection():SendNotification("BasicCommunication.OnEventChanged",
+    { eventName = "AUDIO_SOURCE", isActive = true })
+  m.getMobileSession():ExpectNotification("OnHMIStatus",
+    { hmiLevel = "BACKGROUND", audioStreamingState = "NOT_AUDIBLE", systemContext = pSystemContext })
+end
+
+function m.getVehicleDataGenericError( pReqParams, pHmiResParams )
+  local cid = m.getMobileSession():SendRPC("GetVehicleData", pReqParams)
+  m.getHMIConnection():ExpectRequest("VehicleInfo.GetVehicleData", pReqParams)
+  :Do(function(_, data)
+      m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", pHmiResParams)
+    end)
+  m.getMobileSession():ExpectResponse(cid, { success = false, resultCode = "GENERIC_ERROR" })
+end
+
+function m.ignitionOff()
+  local hmiConnection = actions.hmi.getConnection()
+  local mobileConnection = actions.mobile.getConnection()
+  config.ExitOnCrash = false
+  local timeout = 5000
+  local function removeSessions()
+    for i = 1, actions.mobile.getAppsCount() do
+      actions.mobile.deleteSession(i)
+    end
+  end
+  local event = events.Event()
+  event.matches = function(event1, event2) return event1 == event2 end
+  mobileConnection:ExpectEvent(event, "SDL shutdown")
+  :Do(function()
+    removeSessions()
+    StopSDL()
+    config.ExitOnCrash = true
+  end)
+  hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications", { reason = "SUSPEND" })
+  hmiConnection:ExpectNotification("BasicCommunication.OnSDLPersistenceComplete")
+  :Do(function()
+    hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications",{ reason = "IGNITION_OFF" })
+    for i = 1, actions.mobile.getAppsCount() do
+      actions.mobile.getSession(i):ExpectNotification("OnAppInterfaceUnregistered", { reason = "IGNITION_OFF" })
+    end
+  end)
+  hmiConnection:ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = false })
+  :Times(actions.mobile.getAppsCount())
+  local isSDLShutDownSuccessfully = false
+  hmiConnection:ExpectNotification("BasicCommunication.OnSDLClose")
+  :Do(function()
+    utils.cprint(35, "SDL was shutdown successfully")
+    isSDLShutDownSuccessfully = true
+    mobileConnection:RaiseEvent(event, event)
+  end)
+  :Timeout(timeout)
+  local function forceStopSDL()
```

Comment:
@mvorobio I would recommend to use empty lines in order to simplify where the logical block starts/ends as well as local functions. It is hard to read such a long function. Whitespace is crucial to clean readable code. A blank line (or two) help visually separate out logical blocks of code